editor/app/src/main/java/ca/ubc/eml/soiltopargraphy/editor/ui/main/MainFragment.kt

```diff
@@ -75,9 +90,43 @@ class MainFragment : Fragment() {
             activity!!.invalidateOptionsMenu()
         }
 
+        // Button to add a new flag
+        val addFlag = view.findViewById<View>(R.id.addButton)
+        addFlag.setOnClickListener {
+
+            //Add flag to centre of screen
+
+
+
+        }
+
         return view
     }
 
+    //Google maps
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        mMapView = view.findViewById(R.id.map)
+
+        mMapView.onCreate(null)
+        mMapView.onResume()
+        mMapView.getMapAsync(this)
+    }
+
+    //Google maps
```

Comment:
can you use different comments than line 106 so we can see what each of them does?